src/service/gateways/bitstamp.ts

```diff
@@ -0,0 +1,360 @@
+/// <reference path="../../../typings/tsd.d.ts" />
+/// <reference path="../utils.ts" />
+/// <reference path="../../common/models.ts" />
+/// <reference path="nullgw.ts" />
+
+import Config = require("../config");
+import crypto = require('crypto');
+import WebSocket = require('ws');
+import request = require('request');
+import Models = require("../../common/models");
+import Utils = require("../utils");
+import Interfaces = require("../interfaces");
+import moment = require("moment");
+import util = require("util");
+import _ = require('lodash');
+import Q = require("q");
+
+var shortId = require("shortid");
+
+interface OrderBook {
+    timestamp?: string;        // only present in the http get
+	bids : [string, string][];
+	asks : [string, string][];
+}
+
+interface Ticker {
+	id : string;
+	amount : number;
+	price : number;
+}
+
+interface NewOrderAck {
```

Comment:
I'm not sure if `price`, and `amount` are transmitted as number or strings. Based off their market data API, they're strings. We would need to convert from string to number if that is the case.